src/gcodePlanner.cpp

```diff
@@ -448,6 +448,96 @@ void GCodePlanner::addLinesByOptimizer(Polygons& polygons, GCodePathConfig* conf
     }
 }
 
+int GCodePlanner::spiralizeWallSlice(GCodePathConfig* config, PolygonRef wall, PolygonRef last_wall, int last_wall_seam_vertex_idx)
+{
+    // The spiral has to continue on in an anti-clockwise direction from where the last layer finished, it can't jump backwards
+
+    const int n_points = wall.size();
+
+    // origin is where we start from
+    const Point origin = (last_wall_seam_vertex_idx < 0) ? lastPosition : last_wall[last_wall_seam_vertex_idx];
```

Comment:
Note that in a multithreading context, `lastPosition` is *not* the last position in gcode, but the location of the z-seam you set in the frontend.

You really need to compute all start positions before `FffGcodeWriter::processLayer` is called if you want a nice z seam in spiralize vase mode.

Didn't you do this already in the other branch?